tests/Functional/Features/azure-da_authentication.feature

```diff
@@ -0,0 +1,24 @@
+Feature: When using AzureAD directly
+
+  Scenario: When a user is authenticating a token
+    Given I send an authentication request to "https://azure-mfa.stepup.example.com/saml/sso" without NameID
+    Then I should see "Registration"
+    And I fill in "Email address" with "test-user@institution-b.example.com"
+    When I press "Submit"
+    Given the login with Azure MFA succeeds and the following attributes are released:
+      | name                                                       | value                                             |
+      | urn:mace:dir:attribute-def:mail                            | test-user@institution-b.example.com               |
+      | http://schemas.microsoft.com/claims/authnmethodsreferences | http://schemas.microsoft.com/claims/multipleauthn |
+    Then I should be on "https://azure-mfa.stepup.example.com/saml/sso_return"
+    And the SAML Response should contain element "StatusCode" with attribute "Value" with attribute value "urn:oasis:names:tc:SAML:2.0:status:Success"
+
+  Scenario: Should fail when a authnmethodsreferences attribute is not released
+    Given I send an authentication request to "https://azure-mfa.stepup.example.com/saml/sso" without NameID
+    Then I should see "Registration"
+    And I fill in "Email address" with "test-user@institution-b.example.com"
+    When I press "Submit"
+    Given the login with Azure MFA succeeds and the following attributes are released:
+      | name                                                       | value                                             |
+      | urn:mace:dir:attribute-def:mail                            | test-user@institution-b.example.com               |
+    Then I should be on "https://azure-mfa.stepup.example.com/saml/sso_return"
+    And the SAML Response should contain element "StatusCode" with attribute "Value" with attribute value "urn:oasis:names:tc:SAML:2.0:status:AuthnFailed"
```

Comment:
Thanks for adding the Behat tests! Two feedback points:

1. The filename contains a typo (azure-da instead of azure-ad). This also goes for the other feature that was added.
2. The 'Given' step definition could possibly be made more clear by explicitly indicating this is a direct AD authentication. Maybe by sending the SSO to another endpoint? I'm aware this is only visual improvement. But would aid readability of the tests greatly.